ocaml/xapi/repository.ml

```diff
@@ -510,39 +529,61 @@ let apply_immediate_guidances ~__context ~host ~guidances =
              *)
             ()
         | [RestartDeviceModel] ->
-            restart_device_models ~__context host
+            restart_device_models ~__context ~host
         | [RestartToolstack] ->
             Client.Host.restart_agent ~rpc ~session_id ~host
-        | l when eq_set1 l ->
+        | l when GuidanceSet.eq_set1 l ->
             (* EvacuateHost and RestartToolstack *)
             Client.Host.restart_agent ~rpc ~session_id ~host
-        | l when eq_set2 l ->
+        | l when GuidanceSet.eq_set2 l ->
             (* RestartDeviceModel and RestartToolstack *)
-            restart_device_models ~__context host ;
+            restart_device_models ~__context ~host ;
             Client.Host.restart_agent ~rpc ~session_id ~host
-        | l when eq_set3 l ->
+        | l when GuidanceSet.eq_set3 l ->
             (* RestartDeviceModel and EvacuateHost *)
             (* Evacuating host restarted device models already *)
-            if num_of_hosts = 1 then restart_device_models ~__context host ;
+            if num_of_hosts = 1 then restart_device_models ~__context ~host ;
             ()
-        | l when eq_set4 l ->
+        | l when GuidanceSet.eq_set4 l ->
             (* EvacuateHost, RestartToolstack and RestartDeviceModel *)
             (* Evacuating host restarted device models already *)
-            if num_of_hosts = 1 then restart_device_models ~__context host ;
+            if num_of_hosts = 1 then restart_device_models ~__context ~host ;
             Client.Host.restart_agent ~rpc ~session_id ~host
         | l ->
-            let ref = Ref.string_of host in
+            let host' = Ref.string_of host in
             error
               "Found wrong guidance(s) after applying updates on host \
                ref='%s': %s"
-              ref
+              host'
               (String.concat ";" (List.map Guidance.to_string l)) ;
-            raise Api_errors.(Server_error (apply_guidance_failed, [ref])))
+            raise Api_errors.(Server_error (apply_guidance_failed, [host'])))
   with e ->
-    let ref = Ref.string_of host in
-    error "applying immediate guidances on host ref='%s' failed: %s" ref
+    let host' = Ref.string_of host in
+    error "applying immediate guidances on host ref='%s' failed: %s" host'
       (ExnHelper.string_of_exn e) ;
-    raise Api_errors.(Server_error (apply_guidance_failed, [ref]))
+    raise Api_errors.(Server_error (apply_guidance_failed, [host']))
+
+let apply_pending_guidances ~__context ~host ~guidances =
```

Comment:
The word "apply" is a bit vague, but the function `apply_immediate_guidances` above actually executes the actions suggested by the guidances, whereas this function just records them in the DB. Perhaps call it `set_pending_guidances` to avoid confusion.